guiclient/postCreditMemos.cpp

```
@@ -87,78 +88,179 @@ void postCreditMemos::sPost()
-      return;
+      if (distributeInventory::SeriesCreate(cmitems.value("itemsite_id").toInt(), 
+        cmitems.value("qty").toDouble(), "CM", "RS", cmitems.value("cmitem_id").toInt(), itemlocSeries) < 0)
+      {
+        failedItems.append(creditMemoNumber);
+        errors.append(tr("Failed to create itemlocdist record for item %1")
+          .arg(cmitems.value("item_number").toString()));
+        cmitemFail = true;
+        break;
+      }
+
+      hasControlledItems = true;
     }
-    else if (distributeInventory::SeriesAdjust(postPost.value("result").toInt(), this) == XDialog::Rejected)
+
+    // Don't continue on this credit memo because there was an issue with one of it's line items
+    if (cmitemFail)
+      continue;
+
+    // Distribute detail for the records created above
+    if (hasControlledItems && distributeInventory::SeriesAdjust(itemlocSeries, this, QString(), QDate(),
+      QDate(), true) == XDialog::Rejected)
     {
-      rollback.exec();
-      QMessageBox::information( this, tr("Post Sales Credits"), tr("Transaction Canceled") );
-      return;
+      cleanup.exec();
+      // If it's not the last credit memo, ask user if they want to continue
+      if (creditMemos.at() != creditMemos.size())
+      {
+        if (QMessageBox::question(this,  tr("Post Credit Memo"),
+          tr("Posting distribution detail for credit memo number %1 was cancelled but "
+             "there other credit memos to Post. Continue posting the remaining credit memos?")
+          .arg(creditMemoNumber), 
+          QMessageBox::Yes | QMessageBox::No, QMessageBox::No) == QMessageBox::Yes)
+        {
+          failedItems.append(creditMemoNumber);
+          errors.append(tr("Detail Distribution Cancelled"));
+          continue;
+        }
+        else
+        {
+          failedItems.append(creditMemoNumber);
+          errors.append(tr("Detail Distribution Cancelled"));
+          break;
+        }
+      }
+      
```

There should also be appending to lists and continue here, for if it is the last item. We still want to skip the rest of the operations and still want to report the error.